src/librustc_ast_lowering/item.rs

```diff
@@ -76,6 +76,16 @@ impl<'a> Visitor<'a> for ItemLowerer<'a, '_, '_> {
         }
     }
 
+    fn visit_fn(&mut self, fk: FnKind<'a>, sp: Span, _: NodeId) {
+        match fk {
+            FnKind::Fn(FnCtxt::Foreign, _, sig, _, _) => {
+                self.visit_fn_header(&sig.header);
+                visit::walk_fn_decl(self, &sig.decl);
+            }
```

Comment:
Oh I forgot to ask, can you leave a comment that this is skipping visiting an extraneous body, which is an indication of a parse error?

Reply:
for sure